### 13.4. throw & return

```js
function* process() {
  try {
    yield 9000;
    yield 9001;
    yield 9002;
  } catch (e) {}
}
let it = process();
console.log(it.next().value);
console.log(it.throw("foo"));
console.log(it.next());
// 9000
// {done:true, value:undefined}
// {done:true, value:undefined}
```
```js
function* process() {
  yield 9000;
  yield 9001;
  yield 9002;
}
let it = process();
console.log(it.next().value);
console.log(it.throw("foo"));
console.log(it.next());
// 9000
// Exception: foo (execution terminates)
```
```js
function* process() {
  yield 9000;
  yield 9001;
  yield 9002;
}
let it = process();
console.log(it.next().value);
// Firefox only (currently)
console.log(it.return("foo"));
console.log(it.next());
// 9000
// {value: "foo", done: true}
// {value: undefined, done: true}
```
